### ISSUE TITLE
[clang-tidy] clang-analyzer-deadcode.DeadStores false positive

### ISSUE THREAD

**Issue by sweemer** [llvm/llvm-project#55621]
`clang-analyzer-deadcode.DeadStores` gives a false positive when a store is followed by another store inside a try-catch.

In my example below, if `func` throws then the original store will indeed be read from.

```
$ clang-tidy-15 --version
Ubuntu LLVM version 15.0.0

  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: skylake
```
```
$ cat main.cpp
#include <iostream>
#include <stdexcept>

int func() {
  throw std::exception();
}

int main() {
  int i;
  i = 0;
  try {
    i = func();
  } catch (...) {
  }
  std::cout << i << std::endl;
}
```
```
$ clang-tidy-15 -checks=clang-analyzer-* main.cpp
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "main.cpp"
No compilation database found in /home/sweemer or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
1 warning generated.
/home/sweemer/main.cpp:10:3: warning: Value stored to 'i' is never read [clang-analyzer-deadcode.DeadStores]
  i = 0;
  ^   ~
/home/sweemer/main.cpp:10:3: note: Value stored to 'i' is never read
  i = 0;
  ^   ~
```

**Comment by haoNoQ** [comment]
Ok this is weird, this checker was generally supposed to shut up in presence of exceptions, because we can't build a good source-level CFG for them yet.

In particular, this CFG is very far from correct:

![Screen Shot 2022-05-20 at 11 33 08 PM](https://user-images.githubusercontent.com/2094875/169639185-bf186eda-a1f9-40bc-a5af-0a2d91bd4703.png)

Did somebody re-enable CFG without fixing it?
